spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/Engine.scala

```diff
@@ -340,11 +340,10 @@ object Engine {
       _io = new ThreadPool(core * 50)
     }
 
-    val modelPoolSize: Int = if (engineType == MklBlas) {
-      1
-    } else {
-      core
-    }
+    // for dnn model we should set the pool size to 1 also.
+    // otherwise, it will downgrade the performance and
+    // FIXME make the loss to NaN.
+    val modelPoolSize = 1
```

Comment:
why remove the if ... else ...

Reply:
The thread pool size of all the two cases is 1. We need not the `if else`. If the thread pool size of `_model` more than 1, it will downgrade the performance of Resnet50.